vscode/src/commands/HoverCommandsProvider.ts

```diff
@@ -0,0 +1,158 @@
+import { isCodyIgnoredFile } from '@sourcegraph/cody-shared'
+import { throttle } from 'lodash'
+import * as vscode from 'vscode'
+import { telemetryService } from '../services/telemetry'
+import { telemetryRecorder } from '../services/telemetry-v2'
+import { execQueryWrapper as execQuery } from '../tree-sitter/query-sdk'
+
+/**
+ * Provides clickable commands on hover and handles clicking on commands.
+ * NOTE: Behind the feature flag `cody-hover-commands`.
+ */
+export class HoverCommandsProvider implements vscode.Disposable {
+    private disposables: vscode.Disposable[] = []
+
+    private currentPosition: vscode.Position | undefined
+    private currentDocument: vscode.Uri | undefined
+
+    constructor() {
+        if (!this.isEnabled) {
+            return
+        }
+
+        this.disposables.push(
+            vscode.languages.registerHoverProvider('*', { provideHover: this.onHover.bind(this) }),
```

Comment:
We should only register this for files otherwise it may show in weird places (e.g. the commit message )

<img width="169" alt="image" src="https://github.com/sourcegraph/cody/assets/9516420/c8d93397-9d63-4f6e-96d1-29182b9a91f5">


```suggestion
            vscode.languages.registerHoverProvider({ scheme: 'file' }, { provideHover: this.onHover.bind(this) }),
```

Reply:
Good catch!!

Reply:
@umpox i wasn't able to reproduce this actually, do you have a reproduce steps I can follow?

The hover provider only works inside an editor so i'm wondering if the one from your screenshot is some bug on the VS Code side 🤔

Reply:
Thanks! Are the cody commands showing up on the top of the hover window for you after this change?

Reply:
With this change, the commands are showing up at the top of the hover instead of the bottom:

![image](https://github.com/sourcegraph/cody/assets/68532117/d7ec582e-4c32-43b7-bf5a-0e5c895b75d2)

Updated to move the filtering to the `onHover` method